lib/oli/delivery/sections.ex

```diff
@@ -2420,18 +2422,68 @@ defmodule Oli.Delivery.Sections do
     |> Enum.into(%{})
   end
 
-  defp get_pages(section_slug) do
+  def get_pages(section_slug, params \\ %{}) do
+    text_filter =
+      if params[:text_search] do
+        dynamic([_sr, _s, _spp, _pr, rev], ilike(rev.title, ^"%#{params.text_search}%"))
+      else
+        true
+      end
+
+    limit =
+      if params[:limit] do
+        params.limit
+      else
+        nil
+      end
+
+    offset =
+      if params[:offset] do
+        params.offset
+      else
+        0
+      end
```

Comment:
Maybe in `if` statements you could use the short way of writing it, something like: limit = if params[:limit], do: params.limit, else: nil , WDYT?